schema_based_on_diagram.sql

```diff
@@ -0,0 +1,51 @@
+CREATE TABLE "treatments"(
+    "id" INTEGER NOT NULL,
+    "type" VARCHAR(50) NOT NULL,
+    "name" VARCHAR(50) NOT NULL
+);
+ALTER TABLE
+    "treatments" ADD PRIMARY KEY("id");
+CREATE TABLE "medical_histories"(
+    "id" INTEGER NOT NULL,
+    "admitted_at" TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL,
+    "patient_id" INTEGER NOT NULL,
+    "status" VARCHAR(50) NOT NULL
+);
+ALTER TABLE
+    "medical_histories" ADD PRIMARY KEY("id");
+CREATE TABLE "invoices"(
+    "id" INTEGER NOT NULL,
+    "total_amount" DECIMAL(8, 2) NOT NULL,
+    "generated_at" TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL,
+    "payed_at" TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL,
+    "medical_history_id" INTEGER NOT NULL
+);
+ALTER TABLE
+    "invoices" ADD PRIMARY KEY("id");
+CREATE TABLE "patients"(
+    "id" INTEGER NOT NULL,
+    "name" VARCHAR(50) NOT NULL,
+    "data_of_birth" DATE NOT NULL
+);
+ALTER TABLE
+    "patients" ADD PRIMARY KEY("id");
+CREATE TABLE "invoice_items"(
+    "id" INTEGER NOT NULL,
+    "unit_price" DECIMAL(8, 2) NOT NULL,
+    "quantity" INTEGER NOT NULL,
+    "total_price" DECIMAL(8, 2) NOT NULL,
+    "invoice_id" INTEGER NOT NULL,
+    "treatment_id" INTEGER NOT NULL
+);
+ALTER TABLE
+    "invoice_items" ADD PRIMARY KEY("id");
+ALTER TABLE
+    "treatments" ADD CONSTRAINT "treatments_id_foreign" FOREIGN KEY("id") REFERENCES "medical_histories"("id");
+ALTER TABLE
+    "invoice_items" ADD CONSTRAINT "invoice_items_id_foreign" FOREIGN KEY("id") REFERENCES "treatments"("id");
```

Comment:
- [ ] The **_foreign key_** should be created using the `treatment_id` column in the `invoive_items` table and not the `ID` of the `invoice_items` table based on the schema diagram. This will enable the relationship between tables to be accurately established. Kindly implement this change. 👍 😎